### PR TITLE
Multiplex LSP-compiler over stdout for Toit version.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tools/lsp/server/multiplex.toit [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import host.pipe show OpenPipe
+import monitor show Semaphore
+
+/**
+Connection that dispatches the data of the given $OpenPipe to two pipes.
+The $compiler_to_fs and $compiler_to_parser $SimplePipe should be used as a
+  normal $CloseableReader.
+
+If data is produced faster than it is consumed, then the data is buffered. There
+  is now flow-control.
```
Comment (kasperl) [inline tools/lsp/server/multiplex.toit:15]: now -> no

File: tools/lsp/server/multiplex.toit [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+The given incoming $OpenPipe should be the stdout of the C++ compiler. Data
```
Comment (kasperl) [inline tools/lsp/server/multiplex.toit:17]: So if you ever use printing to debug the compiler, we're going to mess with the protocol?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/lsp/server/compiler.toit [hunk 3]
```diff
@@ -81,32 +82,33 @@ class Compiler:
...
+    multiplex.start_dispatch
+    to_parser := multiplex.compiler_to_parser
+    file_server := PipeFileServer protocol cpp_to multiplex.compiler_to_fs
+    file_server_line := file_server.run
 
     if timeout_ms_ > 0:
       task:: catch --trace:
         sleep --ms=timeout_ms_
         if not has_terminated:
+          print_on_stderr_ "Killing timeout"
```
[the remark this answers is not in the document]
Comment (floitsch) [reply tools/lsp/server/compiler.toit]: my mistake. Was just debugging output.
removed.

File: tools/lsp/server/multiplex.toit (hunk 1 above)
Comment (floitsch) [reply tools/lsp/server/multiplex.toit]: done.

File: tools/lsp/server/multiplex.toit (hunk 2 above)
Comment (floitsch) [reply tools/lsp/server/multiplex.toit:17]: Correct.
That was already the case before, though. The compiler published its diagnostics... on stdout, and used a socket to get filesystem information.
Now it uses stdout for both.

**Pull request merged**